includes/class-affiliatewp-allow-own-referrals.php

```diff
@@ -0,0 +1,328 @@
+		 * time. Also prevents needing to define globals all over the place.
+		 *
+		 * @since 1.0
+		 * @static
+		 *
+		 * @param string $file Path to the main plugin file.
+		 * @return \AffiliateWP_Allow_Own_Referrals The one true bootstrap instance.
+		 */
+		public static function instance( $file = '' ) {
+			// Return if already instantiated.
+			if ( self::is_instantiated() ) {
+				return self::$instance;
+			}
+
+			// Setup the singleton.
+			self::setup_instance( $file );
+
+			self::$instance->setup_constants();
+			self::$instance->load_textdomain();
+			self::$instance->includes();
+			self::$instance->hooks();
+
+			return self::$instance;
+		}
+
+		/**
+		 * Setup the singleton instance
+		 *
+		 * @since 1.0.3
+		 *
+		 * @param string $file File path to the main plugin file.
+		 */
+		private static function setup_instance( $file ) {
+			self::$instance       = new AffiliateWP_Allow_Own_Referrals;
+			self::$instance->file = $file;
+		}
+
+		/**
+		 * Return whether the main loading class has been instantiated or not.
+		 *
+		 * @since 1.0.3
+		 *
+		 * @return bool True if instantiated. False if not.
+		 */
+		private static function is_instantiated() {
+
+			// Return true if instance is correct class
+			if ( ! empty( self::$instance ) && ( self::$instance instanceof AffiliateWP_Allow_Own_Referrals ) ) {
+				return true;
+			}
+
+			// Return false if not instantiated correctly.
+			return false;
+		}
+
+		/**
+		 * Throws an error on object clone.
+		 *
+		 * The whole idea of the singleton design pattern is that there is a single
+		 * object therefore, we don't want the object to be cloned.
+		 *
+		 * @since 1.0.3
+		 *
+		 * @return void
+		 */
+		protected function __clone() {
+			// Cloning instances of the class is forbidden.
+			_doing_it_wrong( __FUNCTION__, __( 'Cheatin&#8217; huh?', 'affiliatewp-allow-own-referrals' ), '1.0' );
+		}
+
+		/**
+		 * Disables unserialization of the class.
+		 *
+		 * @since 1.0.3
+		 *
+		 * @return void
+		 */
+		protected function __wakeup() {
+			// Unserializing instances of the class is forbidden
+			_doing_it_wrong( __FUNCTION__, __( 'Cheatin&#8217; huh?', 'affiliatewp-allow-own-referrals' ), '1.0' );
+		}
+
+		/**
+		 * Sets up the class.
+		 *
+		 * @since 1.0.3
+		 */
+		private function __construct() {
+			self::$instance = $this;
+		}
+
+		/**
+		 * Resets the instance of the class.
+		 *
+		 * @since 1.0.3
+		 * @static
+		 */
+		public static function reset() {
+			self::$instance = null;
+		}
+
+		/**
+		 * Sets up plugin constants.
+		 *
+		 * @since 1.0.3
+		 *
+		 * @return void
+		 */
+		private function setup_constants() {
+			// Plugin version.
+			if ( ! defined( 'AFFWP_AOR_VERSION' ) ) {
+				define( 'AFFWP_AOR_VERSION', $this->version );
+			}
+
+			// Plugin Folder Path.
+			if ( ! defined( 'AFFWP_AOR_PLUGIN_DIR' ) ) {
+				define( 'AFFWP_AOR_PLUGIN_DIR', plugin_dir_path( __FILE__ ) );
```

Comment:
`__FILE__` should be `$this->file` pointing to the main plugin file.
```suggestion
				define( 'AFFWP_AOR_PLUGIN_DIR', plugin_dir_path( $this->file ) );
```